### PR TITLE
Cleanup INTERSECTION_OBSERVER_POLYFILL experiment

### LINKED ISSUE #27807: Deploy a universal IntersectionObserver polyfill in all AMP modes
## Describe the new feature or change to an existing feature you'd like to see

The [standard polyfill](https://github.com/w3c/IntersectionObserver) can be adjusted to work for all AMP cases. We can deploy it and remove all other polyfills and ensure near-universal availability for different measurement tasks, including:

* Async BCR measurement via IntersectionObserver
* Resource scheduling using IntersectionObserver

### PR DESCRIPTION
**Pull request by dvoytenko**
Closes #27807.

TODO

- [x] Resolve "friendly inabox" case.

### CODE REVIEW

File: src/utils/intersection-observer-polyfill.js [hunk 1]
```diff
@@ -14,9 +14,6 @@
  * limitations under the License.
  */
 
-// TODO(#27807): Remove polyfill part InOb polyfill has been launched. But the
```
Comment (zhouyx) [inline src/utils/intersection-observer-polyfill.js:17]: I think it's worth changing the `IntersectionObserverHostApi`, `IntersectionObserverHostForAd` 
 class later on, also the file name to be more clear : )

File: src/polyfills.js [hunk 2]
```diff
@@ -48,16 +48,5 @@ if (self.document) {
   // The anonymous class parameter allows us to detect native classes vs
   // transpiled classes.
   installCustomElements(self, class {});
-  // The AMP and Inabox are launched separately and so there are two
-  // experiment constants.
-  if (
-    // eslint-disable-next-line no-undef
-    INTERSECTION_OBSERVER_POLYFILL ||
```
Comment (zhouyx) [inline src/polyfills.js:55]: Can also remove the flag from experiments-const.json

File: src/utils/intersection-observer-polyfill.js (hunk 1 above)
Comment (dvoytenko) [reply src/utils/intersection-observer-polyfill.js:17]: Sure. But I think it still covers more than ads. It's more of a "legacy 3p intersection observer host support".

**Comment by dvoytenko** [comment]
Ok, the last issue has been resolved. I'm restoring this pull request.

**Comment by zhouyx** [comment]
🎉 🎉  Woohoo! @micajuine-ho fyi

File: extensions/amp-ad/0.1/amp-ad-xorigin-iframe-handler.js [hunk 3]
```diff
@@ -586,9 +586,6 @@ export class AmpAdXOriginIframeHandler {
    * @param {boolean} inViewport
    */
   viewportCallback(inViewport) {
-    if (this.intersectionObserverHost_) {
```
Comment (lannka) [inline extensions/amp-ad/0.1/amp-ad-xorigin-iframe-handler.js:589]: likely the removal of this code might have caused issue: https://github.com/ampproject/amphtml/issues/29759
@dvoytenko was it intended to remove this?

File: extensions/amp-iframe/0.1/amp-iframe.js [hunk 4]
```diff
@@ -596,13 +589,6 @@ export class AmpIframe extends AMP.BaseElement {
     return true;
   }
 
-  /** @override  */
-  viewportCallback(inViewport) {
```
Comment (lannka) [inline extensions/amp-iframe/0.1/amp-iframe.js:600]: according to the offline chat I had with @zhouyx , I think this change is working as intended as amp-iframe does behave differently than amp-ad. @zhouyx pls confirm.

/cc @jridgewell @kristoferbaxter

Comment (zhouyx) [reply extensions/amp-iframe/0.1/amp-iframe.js:600]: Yes I can confirm that `IntersectionObserverHostApi` is working as expected. 
We are using the native IntersectionObserver within the `IntersectionObserverHostApi` so viewportCallback is no longer needed. 

Also for legacy reason, `IntersectionObserverHostApi` sends update whenever there's an intersection change, but `IntersectionObserverHostForAd` sends update whenever there's a position change when the element is within the viewport. 

we should migrate to using native IntersectionObserver to replace `viewportCallback` for `IntersectionObserverHostForAd` as well.

**Pull request merged**